### ISSUE TITLE
App sometimes shows contractID instead of user's display name

### ISSUE THREAD

**Issue by dotmacro** [okTurtles/group-income#2437]
### Problem

The app sometimes shows the contractID instead of the user's display name. 

In testing, this happened on mobile and PWA (but not desktop) when the user's display name started with `.` (EDIT: but that doesn't seem to be the sole cause because it happened with a username & display name that didn't include a `.`. It was in the same group, however.)

PWA:
![image](https://github.com/user-attachments/assets/8374ceb3-43f2-42a5-bee0-932b7b53a8d2)

Mobile:
![image](https://github.com/user-attachments/assets/79b08bc4-b7a8-4bbc-af0b-e8e1009dde12)
(Missing most recent notification, but you can see same name mixup in earlier notification)

Desktop:
![image](https://github.com/user-attachments/assets/d3eb9299-42a3-474c-b9a0-c8c8016394e9)

### Solution

Ensure user's display name is displayed as appropriate.

**Comment by Vayras** [comment]
@dotmacro @taoeffect I would like to work on this, can you please assigne to this

**Comment by taoeffect** [comment]
@Vayras sure, assigned you! Let us know on Slack if you have any questions about this issue.

**Comment by dotmacro** [comment]
(Issue edited above.)

`.` in the user's displayname doesn't seem to be the cause because I can reproduce with a username / display name with no `.`.

The proper username contains only alphabetical characters, display name contains only alphabetical characters and spaces, but app displays as contractID:
![image](https://github.com/user-attachments/assets/9ce2b030-7efc-4680-8008-4c4097f95596)

**Comment by Vayras** [comment]
@taoeffect @dotmacro unable to reproduce this so far, is there a way to reproduce this directly? also if I add members or sign up as a user I'm not allowed to use `.` in the username

**Comment by taoeffect** [comment]
@Vayras it sounds like @dotmacro doesn't know how to reproduce it either...

@dotmacro if you're able to reproduce please post the steps here, as well as what site you reproduced this on.

@Vayras I'm going to unassign you for now because I don't think this is a beginner level issue to solve. You are welcome to keep investigating it though if you'd like, or have a look at some of the other issues.

**Comment by dotmacro** [comment]
@Vayras @taoeffect No, I haven't found a way to reproduce it directly or reliably. 

After signing up with an allowed username, you can change the display name to include a `.`

**Comment by Vayras** [comment]
@taoeffect Sure I'll keep testing this from time to time  while looking for other issues to work on 🚀

**Comment by Vayras** [comment]
@dotmacro Thanks for the info I'll test and see if I can reproduce this
